### PR TITLE
DELTASPIKE-850 Add Weld 3 profile

### PR DESCRIPTION
**Pull request by rsmeral**

### CODE REVIEW

File: deltaspike/parent/code/pom.xml [hunk 1]
```diff
@@ -417,6 +417,139 @@
                 </dependency>
             </dependencies>
         </profile>
+        
+        <profile>
+            <!-- use this profile to compile and test DeltaSpike with JBoss Weld 3.x -->
+            <id>Weld3</id>
```
Comment (johnament) [inline deltaspike/parent/code/pom.xml]: @rsmeral do you know if we can activate it based on the value of weld.version?

Comment (rsmeral) [inline deltaspike/parent/code/pom.xml]: OK, I found a way: weld-core-bom is actually a "depchain" and not really a BOM, so we can keep just one Weld profile and by just importing weld-core-bom we can omit all other dependencies (weld-core/weld-core-impl, etc). I'll update the PR.

**Comment by rsmeral** [comment]
After all, all that's needed to make Weld 3 work is to remove explicit weld-core dependency. All the necessary dependencies for any given release (1.x, 2.x, 3.x) are pulled in through weld-core-bom depchain in Weld profile's dependencyManagement.

**Comment by rsmeral** [comment]
Should not be merged just yet, most likely this will be soon fixed in Weld.

**Comment by rafabene** [comment]
PR merged

**Pull request closed without merging**